File: legacy/app/src/main/java/com/choosemuse/example/libmuse/Listeners/DataListener.java
package com.choosemuse.example.libmuse.Listeners;

import com.choosemuse.example.libmuse.MainActivity;
import com.choosemuse.libmuse.Muse;
import com.choosemuse.libmuse.MuseArtifactPacket;
import com.choosemuse.libmuse.MuseDataListener;
import com.choosemuse.libmuse.MuseDataPacket;

import java.lang.ref.WeakReference;

public class DataListener extends MuseDataListener {
    final WeakReference<MainActivity> activityRef;

    public DataListener(final WeakReference<MainActivity> activityRef) {
        this.activityRef = activityRef;
    }

    @Override
    public void receiveMuseDataPacket(final MuseDataPacket p, final Muse muse) {
        activityRef.get().receiveMuseDataPacket(p, muse);
    }

    @Override
    public void receiveMuseArtifactPacket(final MuseArtifactPacket p, final Muse muse) {
        activityRef.get().receiveMuseArtifactPacket(p, muse);
    }
}

File: app/src/main/java/com/nikitech/robotex/mindcontrol/model/DataListener.kt
package com.nikitech.robotex.mindcontrol.model

import com.choosemuse.libmuse.Muse
import com.choosemuse.libmuse.MuseArtifactPacket
import com.choosemuse.libmuse.MuseDataListener
import com.choosemuse.libmuse.MuseDataPacket
import com.nikitech.robotex.mindcontrol.MainActivity
import java.lang.ref.WeakReference

class DataListener(private val context: MainActivity) : MuseDataListener() {

    override fun receiveMuseDataPacket(p: MuseDataPacket, muse: Muse) {
        context.receiveMuseDataPacket(p, muse)
    }

    override fun receiveMuseArtifactPacket(p: MuseArtifactPacket, muse: Muse) {
        context.receiveMuseArtifactPacket(p, muse)
    }
}
